src/solver/conv_hip_implicit_gemm_v4.cpp

```diff
@@ -359,10 +359,21 @@ static inline ConvSolution GetSolutionBase(const ConvolutionContext& ctx,
         ctx.general_compile_options;
     // clang-format on
 
+    result.construction_params.push_back(construction_parameters);
+
     if(ctx.direction.IsForward() || ctx.direction.IsBackwardData())
         result.invoker_factory = conv::MakeImplGemmDataInvokerFactory(ctx);
+    else
```

Comment:
Why do we have Factory methon in line no. 365  and lambda function in line no. 368. Can we please have it consistent?

Reply:
Not yet. Please look into `conv::MakeImplGemmDataInvokerFactory()`. It is subject for reworking in the future, no time now.

Reply:
The problem is it was excruciating to despaghettify the factory form convolutionocl.cpp when I was making data PR. So I have left it as it was before for future generations.
Lambdas are better, as it is easier to understand, which parts of the invoker execute for which solver, but as I say, I am not familiar enough with the igemm specifically to convert it easily.
I can spend time on that tho if needed.

Reply:
@DrizztDoUrden Let's better spend time for opening issues dedicated to refactoring of new Invokers. Then we'll be able to assign those to Solver authors. Of course, we shall be ready to explain our design. Let me create umbrella ticket right now.